include/gz/math/VolumetricGridLookupField.hh

```diff
@@ -47,7 +47,7 @@ namespace gz
         private: AxisIndex<T> y_indices_by_lon;
 
         private: std::vector<
-          std::vector<std::vector<std::optional<std::size_t>>>> index_table;
+          std::vector<std::vector<std::optional<I>>>> index_table;
```

Comment:
it looks like this is the only breaking change in this PR, so we can split this out to merge before feature freeze if the rest of the changes have not been reviewed

Reply:
See #475